py/mphal.h

```diff
@@ -66,4 +66,11 @@ mp_uint_t mp_hal_ticks_ms(void);
 mp_uint_t mp_hal_ticks_us(void);
 #endif
 
+#ifndef mp_hal_pin_obj_t
+#define mp_hal_pin_obj_t mp_obj_t
+#define mp_hal_get_pin_obj(pin) (pin)
```

Comment:
That's detail of implementation, and we sooner or later will need to apply object type validation. Later is a bit better, as it's the mentioned performance hit. To make it work well, we need to provide a selection of high-level enough pin algorithms (like time_pulse), which validate pin once on entry and then use it many times w/o checking overhead.